numpy/lib/arraypad.py

```diff
@@ -269,17 +209,29 @@ def _get_linear_ramps(padded, axis, width_pair, end_value_pair):
     """
     edge_pair = _get_edges(padded, axis, width_pair)
 
-    left_ramp = _linear_ramp(
-        padded.ndim, axis, start=end_value_pair[0], stop=edge_pair[0],
-        size=width_pair[0], reverse=False
+    left_ramp = np.linspace(
+        start=end_value_pair[0],
+        stop=edge_pair[0],
+        num=width_pair[0],
+        endpoint=False,
+        dtype=padded.dtype,
+        axis=axis,
     )
-    _round_if_needed(left_ramp, padded.dtype)
-
-    right_ramp = _linear_ramp(
-        padded.ndim, axis, start=end_value_pair[1], stop=edge_pair[1],
-        size=width_pair[1], reverse=True
+    # linspace inserts new axis, remove old one with size 1
+    left_ramp = left_ramp.squeeze(axis + 1)
+
+    right_ramp = np.linspace(
+        start=end_value_pair[1],
+        stop=edge_pair[1],
+        num=width_pair[1],
+        endpoint=False,
+        dtype=padded.dtype,
+        axis=axis,
     )
-    _round_if_needed(right_ramp, padded.dtype)
+    # linspace inserts new axis, remove old one with size 1
+    right_ramp = right_ramp.squeeze(axis + 1)
```

Comment:
I don't think the `+1` works correctly if `axis` is negative.

If you squeeze before linspace, I think the problem will go away - `start=end_value_pair[1].squeeze(axis)` etc

Reply:
Actually, I'm now wondering if we should add a `keepdims=True` to linspace, with semantics "replace the `axis`th axis with the new linspace and squeeze". @mhvk, do you have thoughts on this?

Reply:
@eric-wieser - you mean something like the opposite of a reduction, where `keepdims=True` tells `linspace` that there is already a dimension of `1` inside `start` and `stop`? On the one hand, it doesn't quite feel as natural, since in that case one has to check that that dimension is indeed `1`. On the other hand, I can see the nice aspect that then `start` and `stop` would broadcast with the result. Overall, the "other hand" wins for me...

Reply:
Exactly.

Reply:
> If you squeeze before linspace, I think the problem will go away - start=end_value_pair[1].squeeze(axis) etc

That did the trick. I think broadcasting the end_value_pair is not necessary. linspace does that itself when constructing `delta`. Testing seems to confirm that.

Reply:
Opened #14425 from that suggestion